components/list/test/list-item-generic-layout.test.js

```diff
@@ -326,12 +330,23 @@ describe('d2l-list-item-generic-layout', () => {
 				layout = el.querySelector(`[key="${test.itemKey}"]`)
 					.shadowRoot.querySelector('d2l-list-item-generic-layout');
 				test.initial().focus();
-				setTimeout(() => dispatchKeyEvent(layout, test.key.code, test.key.ctrl));
+				let event = null;
+				setTimeout(() => event = dispatchKeyEvent(layout, test.key));
 				await test.event();
 				expect(test.activeElement()).to.equal(test.expected());
+				expect(event.preventDefault).to.have.been.calledOnce;
 			});
 		}
 
+		it('does not preventDefault when Tab is pressed', async() => {
```

Comment:
when I was testing manually, bug symptoms where such: pressing tab does not go move focus outside the grid.
If you can change `longFixture` to have grid and then button, assertions could be made on document.activeElement
In case grid should capture event - the outside of the grid button is not active. For tab key it should be active.
Then you don't need a spy and can do a test in a black-box style.
Does it make sense ?

Reply:
Resolved via zoom

Reply:
as Zina explained earlier. focus will not change properly in headless mode and the suggested test changes will not work